MetadataExtractor/Formats/Photoshop/Subpath.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace MetadataExtractor.Formats.Photoshop
+{
+    /// <summary>
+    /// Represents a subpath created by Photoshop:
+    /// <list type="bullet">
+    /// <item>Closed Bezier knot, linked</item>
+    /// <item>Closed Bezier knot, unlinked</item>
+    /// <item>Open Bezier knot, linked</item>
+    /// <item>Open Bezier knot, unlinked</item>
+    /// </list>
+    /// </summary>
+    /// <author>Payton Garland</author>
+    /// <author>Kevin Mott https://github.com/kwhopper</author>
+    public class Subpath
+    {
+        private List<Knot> p_knots = new List<Knot>();
+
+        public Subpath() : this("")
+        {
+        }
+
+        public Subpath(string type)
+        {
+            Type = type;
+        }
+
+        /// <summary>
+        /// Appends a knot (set of 3 points) into the list
+        /// </summary>
+        /// <param name="knot"></param>
+        public void Add(Knot knot)
+        {
+            p_knots.Add(knot);
+        }
+
+        /// <summary>
+        /// Gets size of knots list
+        /// </summary>
+        /// <returns>size of knots List</returns>
+        public int Size()
```

Comment:
This could be a property. Alternatively, remove this `Size` member, make `GetKnots` an `IReadOnlyList<Knot>` and allow callers to use `subPath.Knots.Count`, which would feel more idiomatic in .NET. The name "size" is more Java-esque.

Reply:
That works, until NET3.5 compat kicks in - then it gets a little weird. I might change the name from GetKnots to Knots and "Size" to "KnotsCount". Otherwise, it has to be the Count() Linq method, which will cause an enumeration. I'll post it and see what you think.